airflow/migrations/versions/788fcbd36a03_add_source_code_table.py

```diff
@@ -0,0 +1,85 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""add source code table
+
+Revision ID: 788fcbd36a03
+Revises: a4c2fd67d16b
+Create Date: 2020-02-20 14:40:23.257645
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import mysql
+
+# revision identifiers, used by Alembic.
+revision = '788fcbd36a03'
+down_revision = 'a4c2fd67d16b'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    """Apply add source code table"""
+    op.create_table('dag_code',  # pylint: disable=no-member
+                    sa.Column('fileloc', sa.String(length=2000), nullable=False),
+                    sa.Column('fileloc_hash', sa.Integer(), nullable=False),
+                    sa.Column('source_code', sa.Text(), nullable=False),
+                    sa.Column('last_updated', sa.DateTime(), nullable=False),
+                    sa.PrimaryKeyConstraint('fileloc', 'fileloc_hash'))
```

Comment:
Is it worth also adding an index on `fileloc`?

Reply:
For now the only query that uses this table to get the code uses fileloc_code_hash column in where.
Besides, correct me if I am wrong, most dbs create an index on primary key by default. but maybe it is better to stay on the saf-side. I  have no strong opinion here.

Reply:
Yes index would be created for all primary keys